### ISSUE TITLE
Review ConnectionInfo / ConnectionDetails

### ISSUE THREAD

**Issue by withakay** [ably/ably-dotnet#231]
Other ably libs just use a `ConnectionDetails` class, the dotnet lib has a `ConnectionInfo` class that has values mapped from a `ConnectionDetails` class (which is in a file called `ConnectionDetailsMessage.cs`).

We should review the rationale behind this design and see if it can be reduced to one class, to better match the other libraries.

If the design does in fact need 2 classes then the file name should at least match the class name, per the style guide.

**Comment by paddybyers** [comment]
I agree that as a general rule, the library types should have the same names as the corresponding IDL types in the spec. If that's not possible (either due to restrictions in the target language, or implementation constraints in the library), then there should still be an obvious and consistent mapping from IDL type names to library type names.
